plotst/axis.typ

```diff
@@ -2,6 +2,27 @@
 // THIS FILE CONTAINS EVERYTHING TO DRAW AND REPRESENT AXES
 //------------------
 
+// range that supports float parameters
+#let float_range(min, max, step: 1) = {
+    if type(min) == "float" or type(max) == "float" or type(step) == "float" {
+      let it = ()
+      it.push(min)
+      if step < 0 {
+        while it.last() + step > max {
+          assert(it.last() + step < it.last(), message: "step size too small to decrease float")
+          it.push(it.last() + step)
+        }
+      } else {
+        while it.last() + step < max {
+          assert(it.last() + step > it.last(), message: "step size too small to increase float")
+          it.push(it.last() + step)
+        }
+      }
+      it
+    } else {
+      range(min, max, step: step)
+    }
+}
```

Comment:
I think this should be moved to the util.typ file. But otherwise, it looks good to me.

Reply:
> I think this should be moved to the util.typ file. But otherwise, it looks good to me.

that caused a circular dependency

Reply:
I changed the util. The circular dependency shouldn't happen now

Reply:
I made the changes for you. Thanks for helping us out ^^ Its greatly appreciated